### PR TITLE
impl: (VM) unify all UTXO types into a single onledger_request

### PR DESCRIPTION
**Pull request by jorgemmsilva**

### CODE REVIEW

File: packages/iscp/requestdata/request_metadata.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
+package requestdata
+
+import (
+	"github.com/iotaledger/hive.go/marshalutil"
+	iotago "github.com/iotaledger/iota.go/v3"
+	"github.com/iotaledger/wasp/packages/iscp"
+	"github.com/iotaledger/wasp/packages/kv/dict"
+)
+
+type RequestMetadata struct {
```
Comment (dessaya) [inline packages/iscp/requestdata/request_metadata.go:10]: Nit: "metadata" usually means "data about the data". I think that a better name for this object should be something like "RequestData" or "RequestParams" or just "Request".

File: packages/iscp/requestdata/requestdata.go [hunk 2]
```diff
@@ -66,57 +63,48 @@ type RequestData interface {
...
 }
 
 type Features interface {
-	TimeLock() (TimeLockOptions, bool)
-	Expiry() (ExpiryOptions, bool)
-	ReturnAmount() (ReturnAmountOptions, bool)
+	TimeLock() *TimeData
+	Expiry() *TimeData
+	ReturnAmount() (uint64, bool)
 	SwapOption() (SwapOptions, bool) // for the new swap
```
Comment (dessaya) [inline packages/iscp/requestdata/requestdata.go:95]: should be called `SwapOptions`

File: packages/iscp/requestdata/requestdata.go [hunk 3]
```diff
@@ -48,7 +45,7 @@ func (t TypeCode) String() string {
 type UTXOMetaData struct {
 	UTXOInput          iotago.UTXOInput
 	MilestoneIndex     uint32
-	MilestoneTimestamp time.Time
+	MilestoneTimestamp uint64
```
Comment (dessaya) [inline packages/iscp/requestdata/requestdata.go:48]: Why not use `time.Time`?
Also, is this a unix timestamp? In seconds, milliseconds, nanoseconds? IMO every timestamp that is not defined as `time.Time` should have a comment specifying the type, or a name like `MilestoneTimestampUnixMillis`

File: packages/iscp/requestdata/requestdata.go [hunk 4]
```diff
@@ -66,57 +63,48 @@ type RequestData interface {
 }
 
 type TimeData struct {
-	ConfirmingMilestoneIndex uint32
-	ConfirmationTime         time.Time // should better be UnixNano ?
+	MilestoneIndex uint32
+	Timestamp      uint64
```
Comment (dessaya) [inline packages/iscp/requestdata/requestdata.go:67]: Same here

**Review by dessaya — APPROVED** [review]

File: packages/iscp/requestdata/request_offledger.go [hunk 5]
```diff
@@ -4,18 +4,17 @@ import (
...
+	iotago "github.com/iotaledger/iota.go/v3"
 	"github.com/iotaledger/wasp/packages/hashing"
 	"github.com/iotaledger/wasp/packages/iscp"
 	"github.com/iotaledger/wasp/packages/iscp/requestdata/placeholders"
 	"github.com/iotaledger/wasp/packages/kv/dict"
 )
 
 type OffLedger struct {
-	chainID        *iscp.ChainID
+	chainID        iscp.ChainID
```
Comment (dessaya) [inline packages/iscp/requestdata/request_offledger.go:17]: Why this change? IIRC we are using `*iscp.ChainID` everywhere.

File: packages/iscp/requestdata/requestdata.go (hunk 3 above)
Comment (jorgemmsilva) [reply packages/iscp/requestdata/requestdata.go:48]: is a unix timestamp in seconds I think.
I initially had it as time.Time, but changed to uint64 after feedback from @lunfardo314 , also this way it matches the same iotago type

File: packages/iscp/requestdata/request_offledger.go (hunk 5 above)
Comment (jorgemmsilva) [reply packages/iscp/requestdata/request_offledger.go:17]: because this was changed :shrug: 
https://github.com/iotaledger/wasp/blob/stardust-vm/packages/iscp/chainid.go#L57

File: packages/iscp/requestdata/requestdata.go (hunk 2 above)
Comment (jorgemmsilva) [reply packages/iscp/requestdata/requestdata.go:95]: Agree, this will likely be removed anyway soon

File: packages/iscp/requestdata/request_metadata.go (hunk 1 above)
Comment (jorgemmsilva) [reply packages/iscp/requestdata/request_metadata.go:10]: requestData is the name of the package and the new interface that we will use in the VM. Request is already used, and there's also UTXOMetadata.

I think RequestMetadata fits because its the request data that is present in the "metadata feature block".

I agree the naming is getting weird, tho
